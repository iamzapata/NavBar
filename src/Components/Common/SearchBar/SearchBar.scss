@import "styles/variables.scss";

.SearchBar {
  width: 100%;
  &.control.has-icons-left {
    .input {
      padding-left: 4em;
    }

    .icon.is-left {
      top: initial;
      left: 1.5rem;
    }
  }

  input {
    background: $color-gray-bg;
    height: 2.8rem;

    &::placeholder {
      color: $color-black-text;
    }
  }
}
