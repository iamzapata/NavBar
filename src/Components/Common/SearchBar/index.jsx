import React from "react"
import "./SearchBar.scss"

const SearchBar = () => (
  <div className="SearchBar control has-icons-left navbar-item">
    <input
      className="input is-size-7"
      type="text"
      placeholder="Find a Pintrest"
    />
    <span className="icon is-small is-left">
      <i className="fa fa fa-search" />
    </span>
  </div>
)

export default SearchBar
