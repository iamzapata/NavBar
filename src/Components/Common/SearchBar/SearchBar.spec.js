import React from "react"
import { render, cleanup } from "react-testing-library"
import SearchBar from "./index"

describe("<SearchBar />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(<SearchBar />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
