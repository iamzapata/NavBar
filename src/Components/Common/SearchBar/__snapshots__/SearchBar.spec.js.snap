// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> Should render self 1`] = `
<div
  class="SearchBar control has-icons-left navbar-item"
>
  <input
    class="input is-size-7 has-text-red"
    placeholder="Find a Pintrest"
    type="text"
  />
  <span
    class="icon is-small is-left"
  >
    <i
      class="fa fa fa-search"
    />
  </span>
</div>
`;
